### PR TITLE
Readme/Docs update

### PR DESCRIPTION
**Pull request by jterapin**
*Description of changes:*
* Updated documentation for more visibility between pages and added more details for clarification 
* Updated Readme to include a Table of Contents, information/examples on Item Operations and Transactions
* Removed License Comments and added `# frozen_string_literal: true` to each file
   * Updated some methods that were modifying string literals
* Updated `aws-record.gemspec` to pull the correct gem version and files
* Updated `version.rb` to pull the correct gem version as well

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,10 +189,82 @@ end
 # unprocessed items can be retried by calling Aws::Record::BatchWrite#execute!
 operation.execute! until operation.complete?
 ```
+---
+### Transactions
```
Comment (alextwoods) [inline README.md:193]: Nice extra docs!

File: lib/aws-record/record/version.rb [hunk 2]
```diff
@@ -1,18 +1,7 @@
...
-# or in the "license" file accompanying this file. This file is distributed on
-# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
-# or implied. See the License for the specific language governing permissions
-# and limitations under the License.
+# frozen_string_literal: true
 
 module Aws
   module Record
-    VERSION = '2.7.0'
+    VERSION = File.read(File.expand_path('../../../../VERSION', __FILE__)).strip
```
Comment (alextwoods) [inline lib/aws-record/record/version.rb:5]: I think we should release a patch version of the gem w/ this fix (so, we should a changelog entry for this)

**Review by alextwoods — APPROVED** [review]
Looks good!

File: lib/aws-record/record/version.rb (hunk 2 above)
Comment (jterapin) [reply lib/aws-record/record/version.rb:5]: Done!

**Pull request merged**